src/sentry/monitors/clock_tasks/check_timeout.py

```diff
@@ -56,11 +56,18 @@ def dispatch_check_timeout(ts: datetime):
 def mark_checkin_timeout(checkin_id: int, ts: datetime):
     logger.info("checkin_timeout", extra={"checkin_id": checkin_id})
 
-    checkin = (
-        MonitorCheckIn.objects.select_related("monitor_environment")
-        .select_related("monitor_environment__monitor")
-        .get(id=checkin_id)
-    )
+    try:
+        checkin = (
+            MonitorCheckIn.objects.select_related("monitor_environment")
+            .select_related("monitor_environment__monitor")
+            .get(id=checkin_id)
+        )
+    except MonitorCheckIn.DoesNotExist:
+        # The check-in may have reached it's retention period
+        #
+        # XXX(epurkhiser): Are there other reasons a check-in may have been
+        # removed between timeout detection and now?
```

Comment:
I think we should be logging this or at least keeping a metric. The main reason I could see it happening is that they deleted the monitor? Retention period seems pretty unlikely tbh

Reply:
Ahh yeah. that makes more sense 👍